src/gpu/intel/ocl/micro_sdpa.cl

```diff
@@ -405,8 +416,22 @@ micro_sdpa(const global KEY_DATA_T *K, const global half *Q,
         cooperative_prefetch_2d_maybe_rem(V, d, k - k0, D_MAX,
                 (ugemm_kq_wg_tile_m * PREFETCH_D_MAX) / D_MAX, ldv, sg_ij,
                 sg_per_wg, SUBGROUP_SIZE, LSC_LDCC_L1C_L3C);
-#endif
 
+#if VAL_SCALES == QUANTIZE_2D
+        /* Prefetch V scales. */
+        cooperative_prefetch_2d_maybe_rem(V_scales, d / VAL_GROUP_SIZE, k - k0,
+                d / VAL_GROUP_SIZE,
```

Comment:
You'll want a constant here to unroll loops inside `cooperative_prefetch_2d_maybe_rem`:
```suggestion
                D_MAX / VAL_GROUP_SIZE,
```